src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php

```diff
@@ -43,15 +43,23 @@ public function __construct(
     ) {
         $this->connections = $registry->getConnectionNames();
         $this->managers = $registry->getManagerNames();
+
+        if (null === $debugDataHolder) {
+            trigger_deprecation('symfony/doctrine-bridge', '6.4', 'Not passing an instance of "%s" as "$debugDataHolder" to "%s()" is deprecated.', DebugDataHolder::class, __METHOD__);
+        }
     }
 
     /**
      * Adds the stack logger for a connection.
      *
      * @return void
+     *
+     * @deprecated since Symfony 6.4, use a DebugDataHolder instead.
      */
     public function addLogger(string $name, DebugStack $logger)
     {
+        trigger_deprecation('symfony/doctrine-bridge', '6.4', '"%s()" is deprecated. Pass an instance of "%s" to the constructor instead.', __METHOD__, DebugDataHolder::class);
```

Comment:
there's a call to this method in DoctrineDataCollectorWithDebugStackTest::createCollector
DoctrineDataCollectorWithDebugStackTest is `@group legacy` but is this legit?

Reply:
DoctrineDataCollectorWithDebugStackTest is covering the legacy case (DebugStack is the legacy logger-based implementation)

Reply:
Exactly, `DoctrineDataCollectorWithDebugStackTest` covers the workflow I'm deprecating here: The constructor is called without the second parameter and instead, `addLogger()` is called to add the deprecated `DebugStack` logger.

In #50571, I'm deleting the whole test.

Reply:
I've added expectations for the deprecation messages to the test you mentioned.